packages/vscode-extension/src/debug/depsChecker/funcToolChecker.ts

```diff
@@ -51,187 +46,210 @@ export class FuncToolChecker implements IDepsChecker {
-  private async hasNPM(): Promise<boolean> {
-    try {
-      await cpUtils.executeCommand(undefined, this._logger, undefined, "npm", "--version");
       return true;
     } catch (error) {
-      // an error indicates no npm
+      this._telemetry.sendEvent(DepsCheckerEvent.npmNotFound);
       return false;
     }
   }
 
-  private async installFuncCoreTools(version: FuncVersion): Promise<void> {
-    if (isWindows()) {
-      await this.installFuncCoreToolsOnWindows(version);
-    } else {
-      await this.installFuncCoreToolsOnUnix(version);
-    }
+  private async cleanup(): Promise<void> {
+    await fs.emptyDir(FuncToolChecker.getDefaultInstallPath());
   }
 
-  private async installFuncCoreToolsOnWindows(version: FuncVersion): Promise<void> {
-    // on Windows, forced install is needed if the func command is broken.
-    await cpUtils.executeCommand(
-      undefined,
-      this._logger,
-      { timeout: timeout },
-      "npm",
-      "install",
-      "-g",
-      "-f",
-      `${funcPackageName}@${version}`
+  private async installFunc(): Promise<void> {
+    await this._telemetry.sendEventWithDuration(
+      DepsCheckerEvent.funcInstallScriptCompleted,
+      async () => {
+        await this._adapter.runWithProgressIndicator(
+          async () => await this.doInstallPortableFunc(FuncVersion.v3)
+        );
+      }
     );
-
-    // delete func.ps1 if exists to workaround the powershell execution policy issue:
-    // https://github.com/npm/cli/issues/470
-    const funcPSScript = await this.getFuncPSScriptPath();
-    if (await fs.pathExists(funcPSScript)) {
-      await this._logger.debug(`deleting func.ps1 from ${funcPSScript}`);
-      await fs.remove(funcPSScript);
-    }
   }
 
-  private async installFuncCoreToolsOnUnix(version: FuncVersion): Promise<void> {
-    const result: cpUtils.ICommandResult = await cpUtils.tryExecuteCommand(
-      undefined,
-      this._logger,
-      { timeout: timeout },
-      "npm",
-      "install",
-      "-g",
-      `${funcPackageName}@${version}`
+  private async doInstallPortableFunc(version: FuncVersion): Promise<void> {
+    await this._logger.info(
+      Messages.startInstallFunctionCoreTool.replace("@NameVersion", displayFuncName)
     );
 
-    const tryInstallfailed: boolean = result.code !== 0;
-    const needAdminPermission: boolean =
-      result.cmdOutputIncludingStderr.includes("permission denied");
-    const command = `npm install -g ${funcPackageName}@${version} --unsafe-perm true`;
-
-    if (tryInstallfailed && needAdminPermission && isMacOS()) {
-      await cpUtils.withTimeout(
-        timeout,
-        cpUtils.execSudo(this._logger, command),
-        "Install func timeout"
+    try {
+      await cpUtils.executeCommand(
+        undefined,
+        this._logger,
+        { timeout: timeout },
+        "npm",
+        "install",
+        "-f",
+        `${funcPackageName}@${version}`,
+        "--prefix",
+        `${FuncToolChecker.getDefaultInstallPath()}`
```

Comment:
`cpUtils` defaults to `{shell: true}` when running command, which may also fail when the parameter (in this case, `FuncToolChecker.getDefaultInstallPath()`) contains spaces. Maybe you need to use `{shell: false}` just like `DotnetChecker`.

Reply:
Added, and {shell: false} also need to command suffix. see `getExecCommand`